test-src/integration_test.js

```diff
@@ -0,0 +1,151 @@
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+const {Artifact} = require('../');
+const {Queue} = require('./mock-queue');
+
+async function writeFileOfSize(size, name) {
+  let bytes = await new Promise((resolve, reject) => {
+    crypto.randomBytes(size, (err, buf) => {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(buf);
```

Comment:
Yes, but it's done consciously because it ensures that execution of the function absolutely cannot continue beyond this point, even if the callbacks `reject` and `resolve` don't work properly.